packages/common/react-components/src/components/List/List.tsx

```diff
@@ -0,0 +1,270 @@
+  );
+};
+
+const PureListItem = forwardRef<ListItemElement, ListItemProps & { id: string }>(
+  (props: ScopedProps<ListItemProps & { id: string }>, forwardedRef) => {
+    const {
+      __scopeSelect,
+      children,
+      selected: propsSelected,
+      defaultSelected,
+      onSelectedChange,
+      id,
+      slots = {}
+    } = props;
+    const { variant, selectable } = useListContext(LIST_NAME, __scopeSelect);
+
+    const [selected = false, setSelected] = useControllableState({
+      prop: propsSelected,
+      defaultProp: defaultSelected,
+      onChange: onSelectedChange
+    });
+
+    const headingId = useId('listItem__heading');
+
+    return (
+      <Collection.ItemSlot id={id} scope={__scopeSelect} selected={selected}>
+        <ListItemProvider scope={__scopeSelect} headingId={headingId}>
+          <Primitive.li
+            {...slots.root}
+            id={id}
+            ref={forwardedRef}
+            aria-labelledby={headingId}
+            {...{
+              ...(selectable && { role: 'option', 'aria-selected': !!selected }),
+              className: 'flex'
+            }}
+          >
+            {variant === 'ordered-draggable' && (
+              <ListItemDragHandle {...slots.dragHandle} className={slots.dragHandle?.className} />
+            )}
+            {selectable && (
+              <ListItemEndcap>
+                <Checkbox
+                  labelId={headingId}
+                  className='mbs-2.5'
+                  {...{ checked: selected, onCheckedChange: setSelected }}
+                />
+              </ListItemEndcap>
+            )}
+            {children}
+          </Primitive.li>
+        </ListItemProvider>
+      </Collection.ItemSlot>
+    );
+  }
+);
+
+const DraggableListItem = forwardRef<ListItemElement, ListItemProps & { id: string }>(
+  (props: ScopedProps<ListItemProps & { id: string }>, forwardedRef) => {
+    const { attributes, listeners, setNodeRef, transform, transition } = useSortable({
+      id: props.id
+    });
+    const ref = useComposedRefs(forwardedRef, setNodeRef) as ComponentPropsWithRef<typeof Primitive.li>['ref'];
+
+    return (
+      <Collection.Slot scope={props.__scopeSelect}>
+        <PureListItem
+          {...props}
+          ref={ref}
+          slots={{
+            root: { style: { transform: CSS.Transform.toString(transform), transition } },
+            dragHandle: { ...listeners, ...attributes }
+          }}
+        />
+      </Collection.Slot>
+    );
+  }
+);
+
+const ListItem = forwardRef<ListItemElement, ListItemProps>((props: ScopedProps<ListItemProps>, forwardedRef) => {
+  const { variant } = useListContext(LIST_NAME, props.__scopeSelect);
+  const listItemId = useId('listItem');
+
+  if (variant === 'ordered-draggable') {
+    return <DraggableListItem {...props} ref={forwardedRef} id={props.id ?? listItemId} />;
+  } else {
+    return <PureListItem {...props} ref={forwardedRef} id={props.id ?? listItemId} />;
+  }
+});
+
+export {
```

Comment:
sort

Reply:
I’ve (tried to) meaningfully group these here, similar to your suggestion about grouping props:

1. Collection component
2. Collection hooks / management functions
3. Item component
4. Components of Items
5. Item hooks